flaml/autogen/agent/human_proxy_agent.py

```diff
@@ -0,0 +1,124 @@
+from .agent import Agent
+from flaml.autogen.code_utils import extract_code, execute_code
+from collections import defaultdict
+
+
+class HumanProxyAgent(Agent):
+    """(Experimental) A proxy agent for human, that can execute code and provide feedback to the other agents."""
+
+    MAX_CONSECUTIVE_AUTO_REPLY = 100  # maximum number of consecutive auto replies (subject to future change)
+
+    def __init__(
+        self,
+        name,
+        system_message="",
+        work_dir=None,
+        human_input_mode="ALWAYS",
+        max_consecutive_auto_reply=None,
+        is_termination_msg=None,
+        **config,
+    ):
+        """
+        Args:
+            name (str): name of the agent
+            system_message (str): system message to be sent to the agent
+            work_dir (str): working directory for the agent
+            human_input_mode (bool): whether to ask for human inputs every time a message is received.
+                Possible values are "ALWAYS", "TERMINATE", "NEVER".
+                (1) When "ALWAYS", the agent prompts for human input every time a message is received.
+                    Under this mode, the conversation stops when the human input is "exit".
+                (2) When "TERMINATE", the agent only prompts for human input only when a termination message is received or
+                    the number of auto reply reaches the max_consecutive_auto_reply.
+                    When human is prompted for input, the conversation stops if there is no input or the input is "exit".
+                (3) When "NEVER", the agent will never prompt for human input. Under this mode, the conversation stops
+                    when the number of auto reply reaches the max_consecutive_auto_reply.
```

Comment:
or when is_termination_msg is True

Reply:
Sure. But need to revise the code also. I revised.